src/main/java/course_project/Main.java

```diff
@@ -0,0 +1,11 @@
+package course_project;
+
+import java.io.IOException;
+
+public class Main {
+
+
+    public static void main(String[] args) throws IOException {
+        Battle.run();
```

Comment:
Quite good! You did PvP seabattle. Looks nice, interface is comfortable. Opt: add delays/enter between players, so they will not see each other fields. Also if ship is sunk, near cells can be marked as hit. But the game is good as-is.

Reply:
Nice abstractions, model/service segregation, the code is easy to read in general, I didn't find any bugs. Approved!

Reply:
it was planned, but had no time left. Thank you for checking. Will work on it later and have a whale of a time with my mates, encouraging them go programming)